toolkit.py

```diff
@@ -0,0 +1,196 @@
+        if vid is None:
+            print("Couldn't load video {}".format(video), file=sys.stderr)
+            return
+        frames = []
+        while(vid.isOpened()):
+            ret, frame = vid.read()
+            if not ret:
+                break
+            (x, y) = self.get_table_metadata(frame)
+            frames.append((x, y))
+        with meta_file as csvfile:
+            for i, frame in enumerate(frames):
+                csv_writer = csv.writer(csvfile, lineterminator='\n')
+                csv_writer.writerow([x[0], x[1], y[0], y[1]])
+        vid.release()
+
+    def write_image_metadata(self, image, meta_file):
+        """
+        Reads given image and gets his metadata which
+        are written to a file
+
+        Args:
+            image(string): Name of image to load
+            meta_file(string): Name of file where metadata will be stored
+        """
+        im = cv.imread(image, cv.CV_LOAD_IMAGE_COLOR)
+        if im is None:
+            print("Couldn't read image file {}".format(image), file=sys.stderr)
+            return
+        (x, y) = self.get_table_metadata(im)
+        with meta_file as csvfile:
+            csv_writer = csv.writer(csvfile, lineterminator='\n')
+            csv_writer.writerow([x[0], x[1], y[0], y[1]])
+
+    video_extension_list = ("mkv", "wmv", "avi", "mp4")
+
+    @timeit
+    def write_metadata(self, input_file, output_file=None):
+        """
+        Main function for getting metadata from images or videos.
+        It checks if format of a file is supported and then calls
+        approperiate function
+
+        Args:
+            input_file(string): Path to file to process
+            output_file(string): Path to output file where metadata
+                                 will be stored
+        """
+        if(output_file is None):
+            file_name = self.__get_file_name(input_file)
+            output_file = open("{}_meta.csv".format(file_name), 'wb+')
+
+        if(input_file.endswith(self.video_extension_list)):
+            self.write_video_metadata(input_file, output_file)
+        elif (imghdr.what(input_file)):
+            self.write_image_metadata(input_file, output_file)
+        else:
+            print("Unrecognized file format", file=sys.stderr)
+
+
+def main(input_file, output_file=None):
+    toolkit = BoardToolkit()
+    for file_name in input_file:
+        toolkit.write_metadata(file_name, output_file)
+
+if __name__ == '__main__':
+    desc = "Command line toolkit for creating board2slides dataset"
+    parser = argparse.ArgumentParser(description=desc)
+    parser.add_argument('filename', metavar='filename', nargs='+',
+                        help='list of videos or images to process')
+    parser.add_argument('-o', '--output-file', nargs='?',
+                        type=argparse.FileType('wb', 0),
+                        help='file where metadata will be saved')
+    args = parser.parse_args()
+    main(args.filename, args.output_file)
```

Comment:
@Foxiy Thanks a ton for doing this! If you decide to go this way you might find [docopt](https://github.com/docopt/docopt) very valuable.